lib/online_order.rb

```diff
@@ -0,0 +1,104 @@
+require 'csv'
+require 'awesome_print'
+require 'money'
+require_relative '../lib/order'
+require_relative '../lib/customer'
+I18n.enforce_available_locales = false
+
+module Grocery
+  class OnlineOrder < Order
+    attr_reader :id, :products, :customer, :status
+
+    def initialize(id, products, customer, status)
+      super(id, products)
+      @customer = customer
+      @status = status
+    end
+
+    def total
+      sum_with_tax_fee = super
+      if sum_with_tax_fee != Money.new(0, "USD")
+        sum_with_tax_fee += Money.new(1000, "USD")
+      end
+      return sum_with_tax_fee
+    end
+
+    def add_product(product_name, product_price)
+      if @status == :pending || @status == :paid
+        true_false = super(product_name, product_price)
+        # true if product added, false if product not added
+        return true_false
+      end
+      # unsure how to make this an argument error instead of a standard error
+      raise StandardError.new("Error: Can't add product. Order status is #{@status}.")
```

Comment:
If you can't add the product, instead just return false like the parent class.  Stick to similar behavior unless you really have to.  Don't make it more complicated.

True in life, true in code.